lib/devise/models/authenticatable.rb

```diff
@@ -253,8 +253,11 @@ def find_or_initialize_with_error_by(attribute, value, error=:invalid) #:nodoc:
 
         # Find or initialize a record with group of attributes based on a list of required attributes.
         def find_or_initialize_with_errors(required_attributes, attributes, error=:invalid) #:nodoc:
-          attributes = attributes.to_unsafe_h.with_indifferent_access if attributes.respond_to? :to_unsafe_h
-          attributes = attributes.slice(*required_attributes).with_indifferent_access
+          attributes = if attributes.respond_to? :permit
+            attributes.slice(*required_attributes).permit!.to_h.with_indifferent_access
```

Comment:
Hi @twalpole, curious why did you choose to test `:permit` before using `:permit!`? Unfortunately I have a case of monkey-patched `Hash` with the `#permit` method in an app, so I'm getting a `NoMethodError` here for `#permit!`. I guess it should be safer to test for `#permit!`?